cmd/gencert/gencert.go

```diff
@@ -0,0 +1,281 @@
+	flag.StringVar(&webhookConfigurationName, "webhook-configuration-name", os.Getenv("WEBHOOK_CONFIGURATION_NAME"), "The name of the ValidatingWebhookConfiguration.")
+	flag.StringVar(&webhookServiceName, "webhook-service-name", os.Getenv("WEBHOOK_SERVICE_NAME"), "The name of the service used to route the webhook.")
+	flag.StringVar(&webhookTlsSecretName, "webhook-tls-secret-name", os.Getenv("WEBHOOK_TLS_SECRET_NAME"), "The name of the tls-type secret used for serving the webhooks.")
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	if namespace == "" {
+		setupLog.Info("No namespace specified, defaulting to operator namespace")
+		raw, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+		if err != nil {
+			setupLog.Error(err, "unable to detect operator namespace")
+			os.Exit(1)
+		}
+		namespace = string(raw)
+	}
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                 scheme,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       vars.GenCertLeaderElectionID,
+		Namespace:              namespace,
+		NewClient: func(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
+			return client.New(config, options)
+		},
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	err = mgr.Add(&GenCertRunner{
+		Client:                   mgr.GetClient(),
+		Logger:                   mgr.GetLogger(),
+		Namespace:                namespace,
+		WebhookConfigurationName: webhookConfigurationName,
+		WebhookServiceName:       webhookServiceName,
+		WebhookTlsSecretName:     webhookTlsSecretName,
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to add gen cert runner")
+		os.Exit(1)
+	}
+
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up ready check")
+		os.Exit(1)
+	}
+
+	setupLog.Info("starting manager", "version", vars.Version)
+	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+		setupLog.Error(err, "problem running manager")
+		os.Exit(1)
+	}
+}
+
+type GenCertRunner struct {
+	Client                   client.Client
+	Logger                   logr.Logger
+	Namespace                string
+	WebhookConfigurationName string
+	WebhookServiceName       string
+	WebhookTlsSecretName     string
+}
+
+func (g *GenCertRunner) NeedLeaderElection() bool {
+	return true
+}
+
+func (g *GenCertRunner) Start(ctx context.Context) error {
+	g.Logger.Info("starting cert renew")
+	for {
+		c, err := g.reconcile(ctx)
+		if err != nil {
+			return err
+		}
+
+		renewIn := g.needsRenewIn(c, 15*24*time.Hour)
```

Comment:
I hope we have a test for this :wink:

Just leave the test runner going for 2 weeks...

Reply:
We have now :)